src/env-inl.h

```diff
@@ -983,10 +983,20 @@ inline v8::MaybeLocal<v8::Object> AllocatedBuffer::ToBuffer() {
 inline v8::Local<v8::ArrayBuffer> AllocatedBuffer::ToArrayBuffer() {
   CHECK_NOT_NULL(env_);
   uv_buf_t buf = release();
+  auto callback = [](void* data, size_t length, void* deleter_data){
+    CHECK_NOT_NULL(deleter_data);
+
+    static_cast<v8::ArrayBuffer::Allocator*>(deleter_data)
+        ->Free(data, length);
```

Comment:
There’s an issue with this that I didn’t realize during review, sorry … see https://bugs.chromium.org/p/v8/issues/detail?id=9908#c10 for details

Reply:
@addaleax should i force-push here or open a new PR?

Reply:
@thangktran I think neither. I believe @addaleax already opened a PR to fix: https://github.com/nodejs/node/pull/30946

Reply:
@Trott i checked it just now and it's actually still there.

Reply:
@trott This here is a V8 API issue, #30946 fixes a bug in our code. So there *is* something we need to do.

I think the 3 options here are:

- Temporarily reverting the changes that add the `v8::ArrayBuffer::Allocator::Free` calls and go back to using the deprecated APIs, effectively blocking the upgrade to 8.0 or 8.1 (not sure which)
- Modifying our embedder API to let embedders and ourselves pass a `std::shared_ptr<v8::ArrayBuffer::Allocator>` that is also passed to V8 and then use a pointer to that instead of a raw `v8::ArrayBuffer::Allocator*` (I don’t like this one, it’s quite a bit of work and complexity with little gain)
- Implementing the V8 API suggestion in the V8 bug linked above and then using that. I think that would be the ideal one, but let’s wait for @ulan’s response in the issue first. @thangktran If this works out and you haven’t contributed to V8 before, I’m happy to help answer any questions – I know their process can be a bit unusual for those who aren’t used to it :)

Reply:
@addaleax : Thank you for offering me help. I'm looking forward to it :D
What is your opinion about @ulan 's alternatives? :
>Before we go ahead with the new constructor, I want to discuss the feasibility of the third option:
>1) Change AllocatedBuffer [1] to allocate and keep std::unique_ptr<v8::BackingStore>.
>2) Add Reallocate method to ArrayBufferAllocator (the default implementation would simply allocate a new buffer).
>3) Add a static Reallocate method to v8::BackingStore that takes a unique_ptr and returns a unique_ptr. Internally it calls Reallocate of the array buffer allocator.

Reply:
@thangktran Thanks for the ping! I’ve commented in the issue. Basically, I’m good with Ulan’s suggestion. :+1: